babel/messages/extract.py

```diff
@@ -138,7 +142,7 @@ def extract_from_dir(dirname=None, method_map=DEFAULT_MAPPING,
     absname = os.path.abspath(dirname)
     for root, dirnames, filenames in os.walk(absname):
         for subdir in dirnames:
-            if subdir.startswith('.') or subdir.startswith('_'):
+            if any(subdir.startswith(chars) for chars in directory_filter):
```

Comment:
starstwith can also accept a tuple of strings to try .. so `if subdir.startswith(directory_filter)` should suffice

Also while at it, `extract_messages` command should be modified to accept the `directory_filter` argument in order to be passed to `extract_from_dir`